exla/lib/exla/mlir/module.ex

```diff
@@ -11,11 +11,22 @@ defmodule EXLA.MLIR.Module do
   alias EXLA.Executable
   alias EXLA.Shape
 
+  @context_key {__MODULE__, :mlir_context}
+  def init do
+    :persistent_term.put(@context_key, EXLA.NIF.new_mlir_context() |> unwrap!())
+  end
```

Comment:
What if we initialize one context per client and then we shove this in the client?

Reply:
That might work. I'll try it out

Reply:
I tried initializing this inside the client at the C++ level and that led to some race conditions which ended in segfaults.

Seeing as different contexts will eat up more threads, I'm not sure there's any gain here outside of the code seeming more isolated.

Reply:
Actually, I just realized that multi-device is broken in this implementation already, so I'll have to figure something else out